api_app/analyzers_manager/file_analyzers/intezer_scan.py

```diff
@@ -33,59 +35,107 @@ def _get_access_token(api_key):
     response_json = response.json()
     token = response_json.get("result", "")
     os.environ["INTEZER_TOKEN"] = token
-    os.environ["INTEZER_TOKEN_DATE"] = get_now_date_only()
+    os.environ["INTEZER_TOKEN_DATE"] = get_now_str()
     return token
 
 
+def get_access_token(api_key):
+    intezer_token = os.environ.get("INTEZER_TOKEN", "")
+    intezer_token_date = os.environ.get("INTEZER_TOKEN_DATE", None)
+    intezer_token_datetime_object = None
+    if intezer_token_date:
+        intezer_token_datetime_object = datetime.strptime(
+            intezer_token_date, "%Y-%m-%d %H:%M:%S"
+        )
+    now = get_now()
+    if (
+        not intezer_token
+        or not intezer_token_datetime_object
+        or (intezer_token_datetime_object < now - timedelta(hours=9))
+    ):
+        intezer_token = _retrieve_access_token_from_intezer(api_key)
+        if not intezer_token:
+            raise AnalyzerRunException("token extraction failed")
+
+    return intezer_token
+
+
+def get_intezer_base_url():
```

Comment:
would be better as a top level constant, `INTEZER_BASE_URL=`.

Reply:
I'll change this